lib/ocpp/v201/charge_point.cpp

```diff
@@ -2564,7 +2564,23 @@ void ChargePoint::handle_reset_req(Call<ResetRequest> call) {
         }
     }
 
-    if (this->callbacks.is_reset_allowed_callback(msg.evseId, msg.type)) {
+    const auto is_reset_allowed = [&]() {
+        if (!this->callbacks.is_reset_allowed_callback(msg.evseId, msg.type)) {
+            return false;
+        }
+
+        if (!msg.evseId.has_value()) {
```

Comment:
This looks a bit strange at first, I would add a comment here. 
Besides that, looks fine.